src/components/cc-kv-hash-input/cc-kv-hash-input.js

```diff
@@ -0,0 +1,277 @@
+        this.value = [{ field: '', value: '' }];
+      }
+    }
+  }
+
+  render() {
+    return html`
+      <div class="elements">
+        <span>${i18n('cc-kv-hash-input.header.field')}</span>
+        <span class="element-last">${i18n('cc-kv-hash-input.header.value')}</span>
+        ${this.value.map(this._renderElement)}
+      </div>
+    `;
+  }
+
+  /**
+   * @param {{field: string, value: string}} element
+   * @param {number} index
+   */
+  _renderElement(element, index) {
+    const len = this.value.length;
+    const isLast = index === len - 1;
+    const isAlone = len === 1;
+
+    return html`
+      <div class="element">
+        <cc-input-text
+          class="input-field"
+          label=${i18n('cc-kv-hash-input.element.field-input')}
+          .value=${element.field}
+          hidden-label
+          data-index=${index}
+          ?disabled=${this.disabled}
+          ?readonly=${this.readonly}
+          @cc-input-text:input=${this._onFieldInput}
+          @cc-input-text:requestimplicitsubmit=${this._onAdd}
+        ></cc-input-text>
+        <cc-input-text
+          label=${i18n('cc-kv-hash-input.element.value-input')}
+          .value=${element.value}
+          hidden-label
+          data-index=${index}
+          ?disabled=${this.disabled}
+          ?readonly=${this.readonly}
+          multi
+          @cc-input-text:input=${this._onValueInput}
+          @cc-input-text:requestimplicitsubmit=${this._onAdd}
+        ></cc-input-text>
+
+        ${!isAlone
+          ? html`
+              <cc-button
+                class="delete-button"
+                .icon=${iconDelete}
+                a11y-name=${i18n('cc-kv-hash-input.element.delete')}
+                data-index=${index}
+                ?disabled=${this.disabled || this.readonly}
+                hide-text
+                danger
+                outlined
+                @cc-button:click=${this._onDelete}
+              ></cc-button>
+            `
+          : ''}
+        ${isLast
+          ? html`
+              <cc-button
+                .icon=${iconAdd}
+                a11y-name=${i18n('cc-kv-hash-input.element.add')}
+                data-index=${index}
+                ?disabled=${this.disabled || this.readonly}
+                hide-text
+                outlined
+                @cc-button:click=${this._onAdd}
+              ></cc-button>
+            `
+          : html`<div></div>`}
+      </div>
+    `;
+  }
+
+  /**
+   * @return {HTMLElement}
+   */
+  _getLastFieldInput() {
+    const all = this.shadowRoot.querySelectorAll('.input-field');
+    return /** @type {HTMLElement} */ (all[all.length - 1]);
+  }
+
+  /**
+   * @param {FocusOptions} [options]
+   */
+  _focusLastFieldInput(options) {
+    this._getLastFieldInput().focus(options);
+  }
+
+  static get styles() {
+    return [
+      // language=CSS
+      css`
+        :host {
+          display: block;
+        }
+
+        .elements {
+          align-items: center;
```

Comment:
This is a good idea. But when value is on a single line, the buttons become misaligned. I would love to have our buttons (when icon only) to be the same height than the input. But for now, it is not the case so I won't apply your proposal even if it would be the right thing to do.

Reply:
Ho that's my bad, you're right. :see_no_evil: 

You can add the following style to fix the alignment but I'm not sure if I like it (see below), I leave the final decision to you:
```css
cc-button {
  position: relative;
  top: 2px;
}
```

The proper solution should use `align-items: baseline;` but I think it requires a `cc-icon` update (I'll note this for later).